### PR TITLE
Interactive list

### PR DESCRIPTION
**Pull request by TemesgenA**
### Features added
- Interactivity module
- A function for the "Clear all completed" button
- Event listener to the checkbox

### CODE REVIEW

File: src/modules/updateComplete.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { toDoListData } from './todo.js';
+
+export default function updateCompletedDisplay() {
+  const checkbox = document.querySelectorAll('.checkbox');
+  const label = document.querySelectorAll('.label');
+  const listItemObject = toDoListData();
+  // eslint-disable-next-line no-restricted-syntax
```
Comment (KDlamini) [inline src/modules/updateComplete.js:7]: when I check some of the tasks as completed and then refresh the page, the previously checked boxes become unchecked.

image

- Kindly look into it and ensure that the boxes remain checked even after page refresh.

File: src/modules/interactive.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    }
+  });
+  setData(duplicate);
+}
+
+export function clearCompletedTodos() {
+  const toDoListDatas = toDoListData().filter((todo) => !todo.completed);
+  resetIndex(toDoListDatas);
+  setData(toDoListDatas);
+}
```
Comment (KDlamini) [inline src/modules/interactive.js:35]: The "clear all completed" tasks button is not working. Currently, nothing happens when I click it.

- Kindly look into it and make sure that it removes all completed tasks from the array when the button is clicked.

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi @TemesgenA 🙋‍♀️,  

Great work so far! 💯 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞 
✔️  Tasks items are saved to localStorage
✔️  Good PR description
✔️  Correct use of Gitflow 
✔️  Commit messages are meaningful
✔️  No linter errors

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Kossi-stack — APPROVED** [review]
## Status: Approved ✔️

**Hi @TemesgenA,**

You did a really great job! 💯 💯💯💯
Congratulations! 🎉 🎉 🎉 🎉 

128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c

You may close this Pull Request and merge the changes
Keep being awesome dev's and good luck with your next milestone!

## To Highlight

✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ README file is descriptive.
✅ Commit messages are meaningful.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding! 👏 👏 👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Pull request merged**